libc/CMakeLists.txt

```diff
@@ -663,13 +663,9 @@ add_library(oelibc STATIC
     ${MUSLSRC}/string/index.c
     ${MUSLSRC}/string/memccpy.c
     ${MUSLSRC}/string/memchr.c
-    ${MUSLSRC}/string/memcmp.c
-    ${MUSLSRC}/string/memcpy.c
     ${MUSLSRC}/string/memmem.c
-    ${MUSLSRC}/string/memmove.c
```

Comment:
For my understanding, are memmove.c & memcmp.c being removed because they were never specified as `no-builtin` anyway? GCC actually [documents several other builtins](https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html) if we want to clean that up (e.g. `mempcpy()`) it feels like we should be consistent about it. 

Otherwise, this list seems like a good reflection of the set of functionality _intended_ in the oelibc. I'd keep them around instead of trimming a specific subset if there's no real binary impact (since these get optimized out by the compiler anyway).

Reply:
@CodeMonkeyLeet 
They were removed since they were being compiled already in oecore as well as here in oelibc, using different compiler flags. It is quite hard to tell which compiled version of these functions were being picked up.

We can retain these functions here but make sure that they are compiled using the same flags in oecore.

Alternatively we can remove them, since our free-standing environment (oecore) provides memcpy, memset, memmove and memcmp provides them

In any case, I will leave them as they are for now; but document why they are being compiled in oecore.